### PR TITLE
Hack to make the renderer own the stage canvas

### PR DESCRIPTION
**Pull request by rschamp**
This is for demonstration purposes, if this is the way we decide, I will make official APIs on the VM and the renderer for checking for a renderer or canvas respectively.  I'm not 100% on adding another wrapper around the canvas, but this seemed like the easiest to read way to do it, and doesn't break any CSS.

@paulkaplan When @cwillisf and I spoke yesterday about the possibility of attaching a new canvas to the renderer, we decided it wouldn't be as clean as keeping the canvas around. The WebGL context stores a lot of state that would need to be maintained, and we would invariably miss something, or it could break down the line as we changed things.

/cc @chrisgarrity 

### Resolves

_What Github issue does this resolve (please include link)?_

Fixes the issue where the stage becomes blank after changing language or switching between player/editor mode.  This was due to the GUI and/or Stage components unmounting and remounting the Renderer's `canvas` element.

### Proposed Changes

_Describe what this Pull Request does_
This moves the Stage's `canvas` ownership from the Stage React component to the Renderer.  Then instead of creating/destroying `canvas` elements in React, append/remove the element from a wrapper component.

### Reason for Changes

_Explain why these changes should be made_
Doing this maintains the state of the `canvas`, and the Renderer's WebGL context.  Therefore we can very quickly remount the canvas, and don't have to recreate the renderer, or the textures stored in the WebGL context.

### Test Coverage

_Please show how you have added tests to cover your changes_
I don't think there's a super great way to test this with an automated test yet. But I improved the "player" playground (player.html), so that you can swap between player and editor mode and see that the state isn't lost.

### Browser Coverage
Check the OS/browser combinations tested (At least 2)

Mac
 * [X] Chrome 
 * [ ] Firefox 
 * [X] Safari
 
Windows
 * [ ] Chrome 
 * [ ] Firefox 
 * [ ] Edge
 
Chromebook
 * [ ] Chrome
 
iPad
* [ ] Safari

Android Tablet
* [ ] Chrome

### CODE REVIEW

**Comment by paulkaplan** [comment]
@rschamp looks good to me! Not sure if you want to wait to merge this to have the APIs in place, or replace them later?

File: src/containers/stage.jsx [hunk 1]
```diff
@@ -365,9 +367,10 @@ class Stage extends React.Component {
             onActivateColorPicker, // eslint-disable-line no-unused-vars
             ...props
         } = this.props;
+        if (!this.canvas) this.canvas = this.setCanvas();
```
Comment (rschamp) [inline src/containers/stage.jsx]: Cruft

**Comment by rschamp** [comment]
@paulkaplan I think it'd be good to merge this as-is and then do the APIs separately, and then update GUI to use them after if that's cool with you.

File: src/containers/stage.jsx [hunk 2]
```diff
@@ -52,16 +51,22 @@ class Stage extends React.Component {
             colorInfo: null,
             question: null
         };
+        if (this.props.vm.runtime.renderer) {
+            this.renderer = this.props.vm.runtime.renderer;
+            this.canvas = this.props.vm.runtime.renderer._gl.canvas;
```
[the remark this answers is not in the document]
Comment (rschamp) [reply src/containers/stage.jsx]: Agreed — I'll do the APIs as a follow-up.

**Pull request merged**